### PR TITLE
use Machine.Name for VM name instead of VSphereMachine.Name

### LINKED ISSUE #623: VSphereMachineTemplate determines node name and hostname but MachineDeployment determines VM name in vSphere
/kind feature

**Describe the solution you'd like**
Right now when you create new nodes for a cluster you are typically configuring a `MachineDeployment` and a `VSphereMachineTemplate`.  The `VSphereMachineTemplate` determines the name of the hostname and then the resulting node, so you're left with something that is a little tough to compare across UIs and APIs.  For example, I created a cluster with 8 nodes and get the following from `kubectl get nodes`:
```
NAME                          STATUS   ROLES    AGE   VERSION
small-workloads-58q77         Ready    <none>   12m   v1.15.4
small-workloads-pjc6c         Ready    <none>   18m   v1.15.4
small-workloads-wzf8c         Ready    <none>   12m   v1.15.4
small-workloads-zll7d         Ready    <none>   18m   v1.15.4
toiling-slug-controlplane-0   Ready    master   45h   v1.15.4
toiling-slug-md-0-rlkx4       Ready    <none>   45h   v1.15.4
toiling-slug-md-2-rhfnb       Ready    <none>   30m   v1.15.4
toiling-slug-md-2-w4lgg       Ready    <none>   32m   v1.15.4
```
Then in my management cluster I will get an entirely different response from `kubectl get machines -l 'cluster.x-k8s.io/cluster-name= toiling-slug'` 
```
NAME                                                    PROVIDERID                                       PHASE
dog-wearing-shoes-machine-deployment-65b8bd58d6-jgjdl   vsphere://4223def8-58c2-f606-f814-746a73cec2a6   running
dog-wearing-shoes-machine-deployment-65b8bd58d6-l4vf7   vsphere://42238bda-f6a4-f96b-f3bf-01e3a5034cd7   running
small-workloads-65b8bd58d6-ppvjk                        vsphere://4223e59a-6501-eaa2-d93f-20aa9426cea5   running
small-workloads-65b8bd58d6-zpw8g                        vsphere://4223f384-15a6-7d6f-b215-6b094ff86b36   running
toiling-slug-controlplane-0                             vsphere://4223bfdd-79b5-06e7-6fd4-16d8950137f5   running
toiling-slug-md-0-5f6769c55f-bbfps                      vsphere://4223d2f4-186d-488c-e57b-378c74cdd326   running
toiling-slug-md-1-db976d98f-5tzz5                       vsphere://4223a08d-1946-e541-d213-8806b4145337   running
toiling-slug-md-1-db976d98f-lxglj                       vsphere://4223f39c-6c4f-f2db-018b-5a833b004d98   running
```

**Anything else you would like to add:**
I might be just confusing the behavior of the sample `KubeadmConfigTemplate` which is setting hte hostname to the `ds.meta_data.hostname`, in which case maybe this is a feature request to change the default behavior to instead find out what the MachineDeployment name is, or to pass that information to the guest VM userdata.

**Environment:**

- Cluster-api-provider-vsphere version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

**What this PR does / why we need it**:
Use `Machine.Name` instead of `VSphereMachine.Name` for VM names in vSphere. That way the hostname and name of the VM are matching. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #623 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
use Machine.Name for VM name instead of VSphereMachine.Name
```

### CODE REVIEW

File: pkg/cloud/vsphere/services/govmomi/service.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (vms *VMService) ReconcileVM(ctx *context.MachineContext) (infrav1.VirtualM
 func (vms *VMService) DestroyVM(ctx *context.MachineContext) (infrav1.VirtualMachine, error) {
 
 	vm := infrav1.VirtualMachine{
-		Name:  ctx.VSphereMachine.Name,
+		Name:  ctx.Machine.Name,
```
Comment (andrewsykim) [inline pkg/cloud/vsphere/services/govmomi/service.go]: Don't think this is forwards compatible. Need to re-evaluate _if_ we decide we want this change at all.

Comment (akutz) [reply pkg/cloud/vsphere/services/govmomi/service.go]: I haven't been able to follow this conversation completely, but the name of an infra provider's machine is different than the CAPI provider's machine? I thought the names of `Cluster/InfraCluster` and `Machine/InfraMachine/KubeadmConfig` were always the same (for each group).

Comment (andrewsykim) [reply pkg/cloud/vsphere/services/govmomi/service.go]: btw, I was wrong earlier. This is forwards compatible cause we destroy VMs by `MachineRef` and not name.

**Pull request merged**